lib/Serializer/functions.php

```diff
@@ -181,7 +181,7 @@ function standardSerializer(Writer $writer, $value): void
                 // This item has a numeric index. We just loop through the
                 // array and throw it back in the writer.
                 standardSerializer($writer, $item);
-            } elseif (is_string($name) && is_array($item) && isset($item['attributes'])) {
+            } elseif (is_array($item) && isset($item['attributes'])) {
```

Comment:
Array keys (name) have to be `int` or `string`. And we tested for `is_int` above. So PHPstan correctly reports that the `is_string` check here is useless.

Reply:
But as a result of looking into this, I don't understand why PHPstan allows the code at line 193, `elseif (is_string($name))`, because we know that `$name` must be a string at that point.
And the `else` at line 198 looks useless. The exception can never be thrown, because PHP only allows array keys of type `int` or `string`, so we can never get to that `InvalidArgumentException`

Reply:
array keys are pretty magic and some auto-conversion are going on in the background (some strings are turned into numbers)

https://3v4l.org/amqMg

Reply:
> array keys are pretty magic and some auto-conversions are going on in the background in php-src (some strings are turned into numbers)

If someone tries to have a key `$someArray['42']` then PHP is going to treat it as `$someArray[42]`, and the code (both before and after this PR) is going to go into the `is_int()` case a few lines up. So that's a "feature" of PHP. And I don't think that there will be any callers who actually want/need to force a string key that is an ASCII decimal number.